spec/models/event/camp_spec.rb

```diff
@@ -593,4 +592,86 @@ def clear_attribute(camp, attr)
     end
   end
 
+  context 'hierarchies, a camp' do
+    subject { events(:bund_camp) }
+
+    it 'may belong to a super_camp' do
+      is_expected.to respond_to :super_camp
+    end
+
+    it 'may have sub_camps' do
+      is_expected.to respond_to :sub_camps
+    end
+
+    it 'can be marked as having sub_camps' do
+      is_expected.to respond_to :allow_sub_camps
+      is_expected.to respond_to :allow_sub_camps=
+    end
+
+    it 'may only be attached to a super-camp that allows it' do
+      super_camp = events(:bund_camp)
+      sub_camp = events(:schekka_camp)
+
+      super_camp.update_attribute(:allow_sub_camps, false)
+      sub_camp.parent_id = super_camp.id
+
+      expect(sub_camp).to_not be_valid
+    end
+
+    context 'allowed to have sub_camps' do
+      before { subject.update_attribute(:allow_sub_camps, true) }
+
+      it 'can have sub_camps' do
+        expect do
+          subject.sub_camps << events(:schekka_camp)
+        end.to change { subject.sub_camps.size }.by 1
+      end
+
+      it 'cannot be deleted if sub_camps are attached' do
+        subject.sub_camps << events(:schekka_camp)
+        expect do
+          subject.destroy
+          is_expected.to_not be_valid
+        end.to_not raise_error
```

Comment:
Verständnisfrage: Wieso darf hier kein error entstehen obwohl in der Validation steht `dependent: :restrict_with_error`? Sprechen die beiden Zeilen von unterschiedlichen Arten von errors (validation error vs exception)?

Reply:
Ja, das sind verschiedene Arten von Error. Du hast aber recht, es liest sich komisch. Das `:dependent` könnte auch `:restrict_with_exception` sein, dann würde die expectation nicht mehr passen.

Reply:
Stimmt, `.to_not raise_exception` ist ein alias dafür: https://relishapp.com/rspec/rspec-expectations/v/2-11/docs/built-in-matchers/raise-error-matcher